src/transactions/simulation/TxSimUtils.cpp

```diff
@@ -212,10 +213,51 @@ mutateScaledAccountID(MuxedAccount& acc, uint32_t partition)
     return key;
 }
 
+PoolID
+getPoolID(LiquidityPoolConstantProductParameters const& params)
+{
+    LiquidityPoolParameters lpp(LIQUIDITY_POOL_CONSTANT_PRODUCT);
+    lpp.constantProduct() = params;
+    return xdrSha256(lpp);
+}
+
+PoolID
+getScaledPoolID(AbstractLedgerTxn& ltx, PoolID const& poolID,
+                UnorderedMap<PoolID, LiquidityPoolParameters>& ctPoolIdToParam,
+                uint32_t partition)
+{
+    auto pool = stellar::loadLiquidityPool(ltx, poolID);
+    if (!pool)
+    {
+        // This pool may be created in the same ledger
+        auto it = ctPoolIdToParam.find(poolID);
+        if (it != ctPoolIdToParam.end())
+        {
+            auto params = it->second;
+            replaceIssuer(params.constantProduct().assetA, partition);
+            replaceIssuer(params.constantProduct().assetB, partition);
+            return xdrSha256(params);
+        }
+
+        // The pool is missing, which means the scaled pool will be missing as
+        // well, so just return the original poolID
+        return poolID;
+    }
+
+    auto& cp = pool.current().data.liquidityPool().body.constantProduct();
+    replaceIssuer(cp.params.assetA, partition);
+    replaceIssuer(cp.params.assetB, partition);
+
+    return getPoolID(cp.params);
+}
+
 void
-generateScaledLiveEntries(std::vector<LedgerEntry>& entries,
-                          std::vector<LedgerEntry> const& oldEntries,
-                          uint32_t partition)
+generateScaledLiveEntries(
+    std::vector<LedgerEntry>& entries,
+    std::vector<LedgerEntry> const& oldEntries,
+    UnorderedMap<PoolID, LiquidityPoolConstantProductParameters> const&
+        mPoolIDToAssets,
```

Comment:
nitpick: naming conversion here and in other places. also the naming for this map varies across the file: here it's poolIDToAsset, but in other places it's poolIDToParam. Using the same name would improve clarity.

Reply:
Good catch. Fixed.